### PR TITLE
`Communication`: Add feature availability list to user documentation

### PR DESCRIPTION
**Pull request by julian-wls**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context

As discussed in the last communication meeting the a list of available features should be made publically available to make it easier for users to compare available features for each platform.

### Description
This PR adds a table to the communication documentation (docs/user/communication.rst) and introduces labels that are placed next to each feature to show what the status of feature is on the web app, iOS app and Android app.
Status labels are AVAILABLE, UNAVAILABLE, WIP, PLANNED and NOT PLANNED. 

### Steps for Testing

Prerequisites:
1 Student

Navigate to [the communication documentation](https://artemis-platform--10015.org.readthedocs.build/en/10015/user/communication.html) and verify that there is a new section called **Communication Features Availability** with a table that explains each status and another table containing a row for each feature.

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

## Summary by CodeRabbit

- **New Features**
	- Introduced a "Communication Features Availability" section detailing feature status across platforms.
	- Added subsections for user, moderator, and instructor features with specific functionalities.
	- Included a "Status explained" subsection for clarity on feature availability.
	- Enhanced document with visual elements, including instructional videos.
	- Updated mobile applications documentation to include a reference link for available communication features on iOS and Android platforms.

### CODE REVIEW

**Review by FelberMartin — DISMISSED** [review]
Looking good! 

It would be good to also mention this section in the [Mobile Applications page](https://docs.artemis.cit.tum.de/user/mobile-applications.html). I think a short info like "For the most up-to-date feature availability status, check [link]" would be enough :D

**Review by PaRangger — DISMISSED** [review]
Looks amazing, great job 🥳

**Review by asliayk — APPROVED** [review]
looks great

**Review by anian03 — CHANGES_REQUESTED** [review]
Looks great overall! I just noticed one inconsistency with the confluence page: On iOS, link formatting is already available (that's also what it says on Confluence) – however here it is listed as planned.

File: docs/user/communication.rst [hunk 1]
```diff
@@ -100,6 +100,238 @@ for multiple links.
...
+| **Markdown Textfield Options**                                                                                                                               |
++------------------------------------------------------+--------------------------------------+--------------------+---------------------+---------------------+
+| Tag other users                                      |                                      | |AVAILABLE|        | |AVAILABLE|         | |AVAILABLE|         |
++------------------------------------------------------+--------------------------------------+--------------------+---------------------+---------------------+
+| Tag channels, lectures and exercises                 |                                      | |AVAILABLE|        | |AVAILABLE|         | |AVAILABLE|         |
++------------------------------------------------------+--------------------------------------+--------------------+---------------------+---------------------+
+| Tag FAQ                                              |                                      | |WIP|              | |WIP|               | |PLANNED|           |
```
Comment (cremertim) [inline docs/user/communication.rst:198]:
```suggestion
| Tag FAQ                                              |                                      | |AVAILABLE|              | |WIP|               | |PLANNED|           |
```

As far as i know this feature is already available

Comment (cremertim) [reply docs/user/communication.rst:198]: Imo, it is more referencing channels/lectures/exercises, not taging them. But i dont have a strong opinion on this one

**Review by cremertim — CHANGES_REQUESTED** [review]
Some minor comments, i would like to hear your opinion on it

File: docs/user/communication.rst [hunk 2]
```diff
@@ -100,6 +100,238 @@ for multiple links.
...
++------------------------------------------------------+--------------------------------------+--------------------+---------------------+---------------------+
+| | Update channel information (name, topic,           | Moderators                           | |AVAILABLE|        | |AVAILABLE|         | |AVAILABLE|         |
+| | description)                                       |                                      |                    |                     |                     |
++------------------------------------------------------+--------------------------------------+--------------------+---------------------+---------------------+
+| Create group chat                                    | Everyone                             | |AVAILABLE|        | |AVAILABLE|         | |AVAILABLE|         |
++------------------------------------------------------+--------------------------------------+--------------------+---------------------+---------------------+
+| Kick users from group chat                           | Members of group chat                | |AVAILABLE|        | |AVAILABLE|         | |AVAILABLE|         |
```
[the remark this answers is not in the document]
Comment (julian-wls) [reply docs/user/communication.rst:306]: Sounds good!

File: docs/user/communication.rst (hunk 1 above)
Comment (julian-wls) [reply docs/user/communication.rst:198]: I think it doesn't really matter. Referencing sounds better though :)

**Review by anian03 — APPROVED** [review]

**Review by asliayk — APPROVED** [review]
reapprove

**Review by eylulnc — APPROVED** [review]
Looks great👌🏻

**Review by FelberMartin — APPROVED** [review]
Re-approve

**Review by PaRangger — APPROVED** [review]
Reapprove

**Review by cremertim — APPROVED** [review]
Reapprove. Thanks for applying my changes!

**Pull request merged**